src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -873,6 +873,121 @@
          :headers {"Content-Type" "application/octet-stream"}
          :body (get-code-content environment code-id file-path)}))))
 
+(defn is-bolt-project?
+  [dir]
+  (or (fs/exists? (str dir "/bolt-project.yaml"))
```

Comment:
I think the *only* configuration we can support is `bolt-project.yaml` and `Boltdir/bolt-project.yaml`. We should not care about bolt.yaml and in the case we have a Boltdir, we should make sure that there is a `Boltdir/bolt-project.yaml`.